### PR TITLE
add more information to quickinfo for imported symbols

### LINKED ISSUE #5515: Import aliases have poor quick-info descriptions
image

### PR DESCRIPTION
**Pull request by jwbay**
Fixes #5515

I took the approach here, more or less -- https://github.com/Microsoft/TypeScript/issues/5515#issuecomment-169756522. We basically print out the root symbol meanings above the import, but with the alias name where it makes sense.

The result:
image

I tried to follow prior art on how things are displayed. This is the current behavior in master for quickinfo on a resolved signature of an imported alias.
image

### CODE REVIEW

**Comment by mhegazy** [comment]
@DanielRosenwasser and @bowdenk7 can i get your input on this?

File: tests/cases/fourslash/quickInfoDisplayPartsInternalModuleAlias.ts [hunk 1]
```diff
@@ -15,33 +15,4 @@
...
-verifyImport("a1", moduleMDisplay);
-verifyImport("a2", moduleMDotM1Display);
-verifyImport("a2", moduleMDotM1Display);
-
-verifyImport("a3", moduleMDisplay, "m2");
-verifyImport("a3", moduleMDisplay, "m2");
-verifyImport("a4", moduleMDotM1Display, "m2");
-verifyImport("a4", moduleMDotM1Display, "m2");
\ No newline at end of file
+verify.baselineQuickInfo();
```
Comment (jwbay) [inline tests/cases/fourslash/quickInfoDisplayPartsInternalModuleAlias.ts]: This test was a bit difficult to decipher, so I swapped it for a baseline. Hope that's alright. If it helps, this is the diff for it against master.

<details>

```diff
diff --git a/tests/baselines/reference/quickInfoDisplayPartsInternalModuleAlias.baseline b/tests/baselines/reference/quickInfoDisplayPartsInternalModuleAlias.baseline
index 6c2047f..c476aba 100644
--- a/tests/baselines/reference/quickInfoDisplayPartsInternalModuleAlias.baseline
+++ b/tests/baselines/reference/quickInfoDisplayPartsInternalModuleAlias.baseline
@@ -1,434 +1,722 @@
 [
   {
     "marker": {
       "fileName": "/tests/cases/fourslash/quickInfoDisplayPartsInternalModuleAlias.ts",
       "position": 66
     },
     "quickInfo": {
       "kind": "alias",
       "kindModifiers": "",
       "textSpan": {
         "start": 66,
         "length": 2
       },
       "displayParts": [
         {
+          "text": "(",
+          "kind": "punctuation"
+        },
+        {
+          "text": "alias",
+          "kind": "text"
+        },
+        {
+          "text": ")",
+          "kind": "punctuation"
+        },
+        {
+          "text": " ",
+          "kind": "space"
+        },
+        {
+          "text": "namespace",
+          "kind": "keyword"
+        },
+        {
+          "text": " ",
+          "kind": "space"
+        },
+        {
+          "text": "a1",
+          "kind": "aliasName"
+        },
+        {
+          "text": "\n",
+          "kind": "lineBreak"
+        },
+        {
           "text": "import",
           "kind": "keyword"
         },
         {
           "text": " ",
           "kind": "space"
         },
         {
           "text": "a1",
           "kind": "aliasName"
         },
         {
           "text": " ",
           "kind": "space"
         },
         {
           "text": "=",
           "kind": "operator"
         },
         {
           "text": " ",
           "kind": "space"
         },
         {
           "text": "m",
           "kind": "moduleName"
         }
       ],
       "documentation": [],
       "tags": []
     }
   },
   {
     "marker": {
       "fileName": "/tests/cases/fourslash/quickInfoDisplayPartsInternalModuleAlias.ts",
       "position": 82
     },
     "quickInfo": {
       "kind": "alias",
       "kindModifiers": "",
       "textSpan": {
         "start": 82,
         "length": 2
       },
       "displayParts": [
         {
+          "text": "(",
+          "kind": "punctuation"
+        },
+        {
+          "text": "alias",
+          "kind": "text"
+        },
+        {
+          "text": ")",
+          "kind": "punctuation"
+        },
+        {
+          "text": " ",
+          "kind": "space"
+        },
+        {
+          "text": "namespace",
+          "kind": "keyword"
+        },
+        {
+          "text": " ",
+          "kind": "space"
+        },
+        {
+          "text": "a1",
+          "kind": "aliasName"
+        },
+        {
+          "text": "\n",
+          "kind": "lineBreak"
+        },
+        {
           "text": "import",
           "kind": "keyword"
         },
         {
           "text": " ",
           "kind": "space"
         },
         {
           "text": "a1",
           "kind": "aliasName"
         },
         {
           "text": " ",
           "kind": "space"
         },
         {
           "text": "=",
           "kind": "operator"
         },
         {
           "text": " ",
           "kind": "space"
         },
         {
           "text": "m",
           "kind": "moduleName"
         }
       ],
       "documentation": [],
       "tags": []
     }
   },
   {
     "marker": {
       "fileName": "/tests/cases/fourslash/quickInfoDisplayPartsInternalModuleAlias.ts",
       "position": 104
     },
     "quickInfo": {
       "kind": "alias",
       "kindModifiers": "",
       "textSpan": {
         "start": 104,
         "length": 2
       },
       "displayParts": [
         {
+          "text": "(",
+          "kind": "punctuation"
+        },
+        {
+          "text": "alias",
+          "kind": "text"
+        },
+        {
+          "text": ")",
+          "kind": "punctuation"
+        },
+        {
+          "text": " ",
+          "kind": "space"
+        },
+        {
+          "text": "namespace",
+          "kind": "keyword"
+        },
+        {
+          "text": " ",
+          "kind": "space"
+        },
+        {
+          "text": "a2",
+          "kind": "aliasName"
+        },
+        {
+          "text": "\n",
+          "kind": "lineBreak"
+        },
+        {
           "text": "import",
           "kind": "keyword"
         },
         {
           "text": " ",
           "kind": "space"
         },
         {
           "text": "a2",
           "kind": "aliasName"
         },
         {
           "text": " ",
           "kind": "space"
         },
         {
           "text": "=",
           "kind": "operator"
         },
         {
           "text": " ",
           "kind": "space"
         },
         {
           "text": "m",
           "kind": "moduleName"
         },
         {
           "text": ".",
           "kind": "punctuation"
         },
         {
           "text": "m1",
           "kind": "moduleName"
         }
       ],
       "documentation": [],
       "tags": []
     }
   },
   {
     "marker": {
       "fileName": "/tests/cases/fourslash/quickInfoDisplayPartsInternalModuleAlias.ts",
       "position": 123
     },
     "quickInfo": {
       "kind": "alias",
       "kindModifiers": "",
       "textSpan": {
         "start": 123,
         "length": 2
       },
       "displayParts": [
         {
+          "text": "(",
+          "kind": "punctuation"
+        },
+        {
+          "text": "alias",
+          "kind": "text"
+        },
+        {
+          "text": ")",
+          "kind": "punctuation"
+        },
+        {
+          "text": " ",
+          "kind": "space"
+        },
+        {
+          "text": "namespace",
+          "kind": "keyword"
+        },
+        {
+          "text": " ",
+          "kind": "space"
+        },
+        {
+          "text": "a2",
+          "kind": "aliasName"
+        },
+        {
+          "text": "\n",
+          "kind": "lineBreak"
+        },
+        {
           "text": "import",
           "kind": "keyword"
         },
         {
           "text": " ",
           "kind": "space"
         },
         {
           "text": "a2",
           "kind": "aliasName"
         },
         {
           "text": " ",
           "kind": "space"
         },
         {
           "text": "=",
           "kind": "operator"
         },
         {
           "text": " ",
           "kind": "space"
         },
         {
           "text": "m",
           "kind": "moduleName"
         },
         {
           "text": ".",
           "kind": "punctuation"
         },
         {
           "text": "m1",
           "kind": "moduleName"
         }
       ],
       "documentation": [],
       "tags": []
     }
   },
   {
     "marker": {
       "fileName": "/tests/cases/fourslash/quickInfoDisplayPartsInternalModuleAlias.ts",
       "position": 149
     },
     "quickInfo": {
       "kind": "alias",
       "kindModifiers": "export",
       "textSpan": {
         "start": 149,
         "length": 2
       },
       "displayParts": [
         {
+          "text": "(",
+          "kind": "punctuation"
+        },
+        {
+          "text": "alias",
+          "kind": "text"
+        },
+        {
+          "text": ")",
+          "kind": "punctuation"
+        },
+        {
+          "text": " ",
+          "kind": "space"
+        },
+        {
+          "text": "namespace",
+          "kind": "keyword"
+        },
+        {
+          "text": " ",
+          "kind": "space"
+        },
+        {
+          "text": "m2",
+          "kind": "moduleName"
+        },
+        {
+          "text": ".",
+          "kind": "punctuation"
+        },
+        {
+          "text": "a3",
+          "kind": "aliasName"
+        },
+        {
+          "text": "\n",
+          "kind": "lineBreak"
+        },
+        {
           "text": "import",
           "kind": "keyword"
         },
         {
           "text": " ",
           "kind": "space"
         },
         {
           "text": "m2",
           "kind": "moduleName"
         },
         {
           "text": ".",
           "kind": "punctuation"
         },
         {
           "text": "a3",
           "kind": "aliasName"
         },
         {
           "text": " ",
           "kind": "space"
         },
         {
           "text": "=",
           "kind": "operator"
         },
         {
           "text": " ",
           "kind": "space"
         },
         {
           "text": "m",
           "kind": "moduleName"
         }
       ],
       "documentation": [],
       "tags": []
     }
   },
   {
     "marker": {
       "fileName": "/tests/cases/fourslash/quickInfoDisplayPartsInternalModuleAlias.ts",
       "position": 165
     },
     "quickInfo": {
       "kind": "alias",
       "kindModifiers": "export",
       "textSpan": {
         "start": 165,
         "length": 2
       },
       "displayParts": [
         {
+          "text": "(",
+          "kind": "punctuation"
+        },
+        {
+          "text": "alias",
+          "kind": "text"
+        },
+        {
+          "text": ")",
+          "kind": "punctuation"
+        },
+        {
+          "text": " ",
+          "kind": "space"
+        },
+        {
+          "text": "namespace",
+          "kind": "keyword"
+        },
+        {
+          "text": " ",
+          "kind": "space"
+        },
+        {
+          "text": "m2",
+          "kind": "moduleName"
+        },
+        {
+          "text": ".",
+          "kind": "punctuation"
+        },
+        {
+          "text": "a3",
+          "kind": "aliasName"
+        },
+        {
+          "text": "\n",
+          "kind": "lineBreak"
+        },
+        {
           "text": "import",
           "kind": "keyword"
         },
         {
           "text": " ",
           "kind": "space"
         },
         {
           "text": "m2",
           "kind": "moduleName"
         },
         {
           "text": ".",
           "kind": "punctuation"
         },
         {
           "text": "a3",
           "kind": "aliasName"
         },
         {
           "text": " ",
           "kind": "space"
         },
         {
           "text": "=",
           "kind": "operator"
         },
         {
           "text": " ",
           "kind": "space"
         },
         {
           "text": "m",
           "kind": "moduleName"
         }
       ],
       "documentation": [],
       "tags": []
     }
   },
   {
     "marker": {
       "fileName": "/tests/cases/fourslash/quickInfoDisplayPartsInternalModuleAlias.ts",
       "position": 194
     },
     "quickInfo": {
       "kind": "alias",
       "kindModifiers": "export",
       "textSpan": {
         "start": 194,
         "length": 2
       },
       "displayParts": [
         {
+          "text": "(",
+          "kind": "punctuation"
+        },
+        {
+          "text": "alias",
+          "kind": "text"
+        },
+        {
+          "text": ")",
+          "kind": "punctuation"
+        },
+        {
+          "text": " ",
+          "kind": "space"
+        },
+        {
+          "text": "namespace",
+          "kind": "keyword"
+        },
+        {
+          "text": " ",
+          "kind": "space"
+        },
+        {
+          "text": "m2",
+          "kind": "moduleName"
+        },
+        {
+          "text": ".",
+          "kind": "punctuation"
+        },
+        {
+          "text": "a4",
+          "kind": "aliasName"
+        },
+        {
+          "text": "\n",
+          "kind": "lineBreak"
+        },
+        {
           "text": "import",
           "kind": "keyword"
         },
         {
           "text": " ",
           "kind": "space"
         },
         {
           "text": "m2",
           "kind": "moduleName"
         },
         {
           "text": ".",
           "kind": "punctuation"
         },
         {
           "text": "a4",
           "kind": "aliasName"
         },
         {
           "text": " ",
           "kind": "space"
         },
         {
           "text": "=",
           "kind": "operator"
         },
         {
           "text": " ",
           "kind": "space"
         },
         {
           "text": "m",
           "kind": "moduleName"
         },
         {
           "text": ".",
           "kind": "punctuation"
         },
         {
           "text": "m1",
           "kind": "moduleName"
         }
       ],
       "documentation": [],
       "tags": []
     }
   },
   {
     "marker": {
       "fileName": "/tests/cases/fourslash/quickInfoDisplayPartsInternalModuleAlias.ts",
       "position": 213
     },
     "quickInfo": {
       "kind": "alias",
       "kindModifiers": "export",
       "textSpan": {
         "start": 213,
         "length": 2
       },
       "displayParts": [
         {
+          "text": "(",
+          "kind": "punctuation"
+        },
+        {
+          "text": "alias",
+          "kind": "text"
+        },
+        {
+          "text": ")",
+          "kind": "punctuation"
+        },
+        {
+          "text": " ",
+          "kind": "space"
+        },
+        {
+          "text": "namespace",
+          "kind": "keyword"
+        },
+        {
+          "text": " ",
+          "kind": "space"
+        },
+        {
+          "text": "m2",
+          "kind": "moduleName"
+        },
+        {
+          "text": ".",
+          "kind": "punctuation"
+        },
+        {
+          "text": "a4",
+          "kind": "aliasName"
+        },
+        {
+          "text": "\n",
+          "kind": "lineBreak"
+        },
+        {
           "text": "import",
           "kind": "keyword"
         },
         {
           "text": " ",
           "kind": "space"
         },
         {
           "text": "m2",
           "kind": "moduleName"
         },
         {
           "text": ".",
           "kind": "punctuation"
         },
         {
           "text": "a4",
           "kind": "aliasName"
         },
         {
           "text": " ",
           "kind": "space"
         },
         {
           "text": "=",
           "kind": "operator"
         },
         {
           "text": " ",
           "kind": "space"
         },
         {
           "text": "m",
           "kind": "moduleName"
         },
         {
           "text": ".",
           "kind": "punctuation"
         },
         {
           "text": "m1",
           "kind": "moduleName"
         }
       ],
       "documentation": [],
       "tags": []
     }
   }
 ]
```

</details>

**Comment by DanielRosenwasser** [comment]
Sorry for the delay! I think it's an improvement, but I could see where it'd go wrong for much more complex types. That said, I don't think that should block the change.

**Comment by mhegazy** [comment]
@jwbay sorry for the delay. can you please refresh the PR, and address the failing CI test?

**Comment by mhegazy** [comment]
merged manually. thanks.

**Pull request merged**